edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/Gyro/src/Robot.cpp

```diff
@@ -1,81 +1,42 @@
 #include <AnalogGyro.h>
-#include <CANTalon.h>
 #include <Joystick.h>
 #include <RobotDrive.h>
-#include <SampleRobot.h>
+#include <IterativeRobot.h>
+#include <math.h>
 
 /**
  * This is a sample program to demonstrate how to use a gyro sensor to make a robot drive
  * straight. This program uses a joystick to drive forwards and backwards while the gyro
  * is used for direction keeping.
- *
- * WARNING: While it may look like a good choice to use for your code if you're inexperienced,
- * don't. Unless you know what you are doing, complex code will be much more difficult under
- * this system. Use IterativeRobot or Command-Based instead if you're new.
  */
-class Robot: public SampleRobot {
+class Robot: public IterativeRobot {
 public:
-	/**
-	 * Runs during autonomous.
-	 */
-	void Autonomous() {}
-
-	/**
-	 * Sets the gyro sensitivity and drives the robot when the joystick is
-	 * pushed. The motor speed is set from the joystick while the RobotDrive
-	 * turning value is assigned from the error between the setpoint and the
-	 * gyro angle.
-	 */
-	void OperatorControl() {
-		double turningValue;
-
-		// Calibrates gyro values to equal degrees
-		gyro.SetSensitivity(voltsPerDegreePerSecond);
-
-		while (IsOperatorControl() && IsEnabled()) {
-			turningValue = (angleSetpoint - gyro.GetAngle()) * pGain;
-			if (joystick.GetY() <= 0) {
-				// Forwards
-				myRobot.Drive(joystick.GetY(), turningValue);
-			}
-			else {
-				// Backwards
-				myRobot.Drive(joystick.GetY(), -turningValue);
-			}
-		}
+	void robotInit() {
+		gyro.SetSensitivity(VOLTS_PER_DEGREE_PER_SECOND);
 	}
 
 	/**
-	 * Runs during test mode.
+	 * The motor speed is set from the joystick while the RobotDrive turning
+	 * value is assigned from the error between the setpoint and the gyro angle.
 	 */
-	void Test() {}
+	void teleopPeriodic() {
+		double turningValue = (ANGLE_SETPOINT - gyro.GetAngle()) * KP;
+		// Invert the direction of the turn if we are going backwards
+		turningValue = std::copysign(turningValue, joystick.GetY());
+		myRobot.Drive(joystick.GetY(), turningValue);
+	}
 
 private:
-	constexpr int gyroChannel = 0;  // Analog input
-	constexpr int joystickChannel = 0;  // USB number in DriverStation
-
-	// Channels for motors
-	constexpr int leftFontMotorChannel = 1;
-	constexpr int rightFrontMotorChannel = 0;
-	constexpr int leftRearMotorChannel = 3;
-	constexpr int rightRearMotorChannel = 2;
-
-	double angleSetpoint = 0.0;
-	constexpr double pGain = 0.006;  // Propotional turning constant
+	static constexpr double ANGLE_SETPOINT = 0.0;
+	static constexpr double KP = 0.005;  // Propotional turning constant
```

Comment:
Comment typo and name should be `kP` according to the style guide. An alternative used in control theory often is `Kp`.